File: fixtures/scale/swift/large/Families/Bacon/BaconRunner.swift
// Correctness entry point for the bacon family; wraps CorrectnessEngine
// with this family's cipher and known-answer vectors.
struct BaconRunner {
    private let engine = CorrectnessEngine()

    func check() -> VectorOutcome {
        let cipher = BaconCipher()
        let vectors = BaconVectors.all()
        return engine.verify(cipher, vectors)
    }
}

File: fixtures/scale/php/large/src/Families/Bacon/BaconRunner.php
<?php

namespace CipherLab\Families\Bacon;

use CipherLab\Abstractions\VectorOutcome;
use CipherLab\Engine\CorrectnessEngine;

// Correctness entry point for the bacon family; wraps CorrectnessEngine
// with this family's cipher and known-answer vectors.
class BaconRunner
{
    private CorrectnessEngine $engine;

    public function __construct()
    {
        $this->engine = new CorrectnessEngine();
    }

    public function check(): VectorOutcome
    {
        return $this->engine->verify(new BaconCipher(), BaconVectors::all());
    }
}
